python/packages/core/agent_framework/_tools.py

```diff
@@ -84,7 +85,8 @@
 
 logger = get_logger()
 FUNCTION_INVOKING_CHAT_CLIENT_MARKER: Final[str] = "__function_invoking_chat_client__"
-DEFAULT_MAX_ITERATIONS: Final[int] = 10
+DEFAULT_MAX_ITERATIONS: Final[int] = 40
```

Comment:
Why do we need to change to 40?

Reply:
this is also the default in MEAI, and with this generation of models 10 is probably too low